packages/loopring_v2/contracts/helper/RingHelper.sol

```diff
@@ -340,51 +340,93 @@ library RingHelper {
         pure
         returns (uint)
     {
-        uint buyerFeeAmountAfterRebateB = prevP.feeAmountB.sub(prevP.rebateB);
-
         // If the buyer needs to pay fees in tokenB, the seller needs
         // to send the tokenS amount to the fee holder contract
         uint amountSToBuyer = p.fillAmountS
             .sub(p.feeAmountS)
-            .sub(buyerFeeAmountAfterRebateB);
+            .sub(prevP.feeAmountB.sub(prevP.rebateB)); // buyer fee amount after rebate
 
         uint amountSToFeeHolder = p.feeAmountS
             .sub(p.rebateS)
-            .add(buyerFeeAmountAfterRebateB);
+            .add(prevP.feeAmountB.sub(prevP.rebateB)); // buyer fee amount after rebate
 
         uint amountFeeToFeeHolder = p.feeAmount
             .sub(p.rebateFee);
 
+
         if (p.order.tokenS == p.order.feeToken) {
             amountSToFeeHolder = amountSToFeeHolder.add(amountFeeToFeeHolder);
             amountFeeToFeeHolder = 0;
         }
 
         // Transfers
-        ctx.transferPtr = addTokenTransfer(
-            ctx.transferData,
-            ctx.transferPtr,
-            p.order.feeToken,
-            p.order.owner,
-            address(ctx.feeHolder),
-            amountFeeToFeeHolder
-        );
-        ctx.transferPtr = addTokenTransfer(
-            ctx.transferData,
-            ctx.transferPtr,
-            p.order.tokenS,
-            p.order.owner,
-            address(ctx.feeHolder),
-            amountSToFeeHolder
-        );
-        ctx.transferPtr = addTokenTransfer(
-            ctx.transferData,
-            ctx.transferPtr,
-            p.order.tokenS,
-            p.order.owner,
-            prevP.order.tokenRecipient,
-            amountSToBuyer
-        );
+        if (p.order.broker == address(0x0)) {
+            ctx.transferPtr = addTokenTransfer(
+                ctx.transferData,
+                ctx.transferPtr,
+                p.order.feeToken,
+                p.order.owner,
+                address(ctx.feeHolder),
+                amountFeeToFeeHolder
+            );
+            ctx.transferPtr = addTokenTransfer(
+                ctx.transferData,
+                ctx.transferPtr,
+                p.order.tokenS,
+                p.order.owner,
+                address(ctx.feeHolder),
+                amountSToFeeHolder
+            );
+            ctx.transferPtr = addTokenTransfer(
+                ctx.transferData,
+                ctx.transferPtr,
+                p.order.tokenS,
+                p.order.owner,
+                prevP.order.tokenRecipient,
+                amountSToBuyer
+            );
+        } else {
+            
+            // Calculates amount received from other participant
+            uint amountBFromSeller = prevP.fillAmountS
```

Comment:
@Brechtpd Is this the correct way to calculate how much this participation is receiving from the other? We brought the max participants in a ring down to 2, so I figured taking the logic to calculate how much this participant and sending the other and flipping it would do the trick

Reply:
I think so yes. Using the other order's `amountSToBuyer` is the amount the order actually receives, which seems to be correctly calculated here.